### PR TITLE
Fix TextEdit Select crash

### LINKED ISSUE #29538: Editor crash when using TextEdit::select
**Godot version:**
3.2.dev.custom_build. 197b65f32

**OS/device including version:**
Ubuntu 19.04

**Issue description:**
```
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x43f60) [0x7faeeb3eaf60] (??:0)
[2] CowData<TextEdit::Text::Line>::get(int) const (/home/rafal/Pulpit/godot/./core/cowdata.h:151 (discriminator 7))
[3] Vector<TextEdit::Text::Line>::operator[](int) const (/home/rafal/Pulpit/godot/./core/vector.h:85)
[4] TextEdit::Text::operator[](int) const (/home/rafal/Pulpit/godot/scene/gui/text_edit.h:130)
[5] TextEdit::select(int, int, int, int) (/home/rafal/Pulpit/godot/scene/gui/text_edit.cpp:4837)
[6] TextEdit::undo() (/home/rafal/Pulpit/godot/scene/gui/text_edit.cpp:5617)
[7] TextEdit::menu_option(int) (/home/rafal/Pulpit/godot/scene/gui/text_edit.cpp:6396)
[8] MethodBind1<int>::call(Object*, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/./core/method_bind.gen.inc:775 (discriminator 12))
[9] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/core/object.cpp:940 (discriminator 1))
[10] Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) (/home/rafal/Pulpit/godot/core/variant_call.cpp:1071 (discriminator 1))
[11] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript_function.cpp:1078)
[12] GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) (/home/rafal/Pulpit/godot/modules/gdscript/gdscript.cpp:1192)
[13] Node::_notification(int) (/home/rafal/Pulpit/godot/scene/main/node.cpp:58)
[14] Node::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/main/node.h:46 (discriminator 14))
[15] CanvasItem::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/2d/canvas_item.h:166 (discriminator 3))
[16] Control::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/gui/control.h:51 (discriminator 3))
[17] TextEdit::_notificationv(int, bool) (/home/rafal/Pulpit/godot/scene/gui/text_edit.h:43 (discriminator 3))
[18] Object::notification(int, bool) (/home/rafal/Pulpit/godot/core/object.cpp:952)
[19] SceneTree::_notify_group_pause(StringName const&, int) (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:975)
[20] SceneTree::idle(float) (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:522 (discriminator 3))
[21] Main::iteration() (/home/rafal/Pulpit/godot/main/main.cpp:1919)
[22] OS_X11::run() (/home/rafal/Pulpit/godot/platform/x11/os_x11.cpp:3034)
[23] /usr/bin/godot(main+0xdc) [0x138598e] (/home/rafal/Pulpit/godot/platform/x11/godot_x11.cpp:56)
[24] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb) [0x7faeeb3cdb6b] (??:0)
[25] /usr/bin/godot(_start+0x2a) [0x13857fa] (??:?)
```

**Steps to reproduce:**
1. Run minimal project and wait

**Minimal reproduction project:**
https://github.com/qarmin/The-worst-Godot-test-project
commit d7c2e29cce8f1f2ee0d04a2b528f00faa6f809d7
ZIP - https://github.com/qarmin/The-worst-Godot-test-project/archive/d7c2e29cce8f1f2ee0d04a2b528f00faa6f809d7.zip

### PR DESCRIPTION
**Pull request by qarmin**
Fix #29538

### CODE REVIEW

**Comment by KoBeWi** [comment]
`p_from_line` also causes crash when negative. You should add it to the condition too.

File: scene/gui/text_edit.cpp [hunk 1]
```diff
@@ -4825,6 +4825,9 @@ void TextEdit::deselect() {
 
 void TextEdit::select(int p_from_line, int p_from_column, int p_to_line, int p_to_column) {
 
+	ERR_FAIL_COND(p_to_line < 0)
+	ERR_FAIL_COND(p_from_line < 0)
```
Comment (akien-mga) [inline scene/gui/text_edit.cpp]: I'm surprised this compiles, but `ERR_FAIL_COND()` macros should end with a semicolon.

Comment (akien-mga) [reply scene/gui/text_edit.cpp]: I wonder why the `_column` parameters have checks resetting them to 0 below, but not the `_line` parameters though. CC @Paulb23

Comment (Paulb23) [reply scene/gui/text_edit.cpp]: I don't think there's any particular reason, resetting them to `0` would be fine.

Comment (akien-mga) [reply scene/gui/text_edit.cpp]: Bump.

Comment (qarmin) [reply scene/gui/text_edit.cpp]: Done

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**